kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/ResolverAAImpl.kt

```diff
@@ -565,16 +560,30 @@ class ResolverAAImpl(
         return visitor.symbols
     }
 
-    private val deferredSymbolsRestored: Set<KSAnnotated> by lazy {
-        deferredSymbols.values.flatten().mapNotNull { it.restore() }.toSet()
+    private val annotationToSymbolsMap: Map<String, Collection<KSAnnotated>> by lazy {
+        mapAnnotatedSymbols(false)
+    }
+
+    private val annotationToSymbolsMapWithLocals: Map<String, Collection<KSAnnotated>> by lazy {
+        mapAnnotatedSymbols(true)
     }
 
-    private val newAnnotatedSymbols: Collection<KSAnnotated> by lazy {
-        collectAnnotatedSymbols(false)
+    private fun mapAnnotatedSymbols(inDepth: Boolean): Map<String, Collection<KSAnnotated>> {
+        val newSymbols = collectAnnotatedSymbols(inDepth)
+        val withDeferred = newSymbols + deferredSymbolsRestored
+        return mutableMapOf<String, MutableCollection<KSAnnotated>>().apply {
+            withDeferred.forEach { annotated ->
```

Comment:
Nit: http://go/kos-rule/DiscourageForEach (internal link)

Instead of `...forEach { ... return@forEach }`, just regular `for` loop and `continue` is easy to read (at least to me...)

Reply:
Done.